lib/ingram_micro/elements/outbound_shipment_information.rb

```diff
@@ -1,54 +1,46 @@
 class IngramMicro::OutboundShipmentInformation < IngramMicro::BaseElement
-
-  SHIPPING_METHODS = {
-    'FX01' => 'FedEx Standard Overnight',
-    'FXAM' => 'FedEx Priority Overnight',
-    'FX2D' => 'FedEx Second Day',
-    'FXSP' => 'FedEx Smart Post',
-    'FXIE' => 'FedEx International Economy',
-    'FXIP' => 'FedEx International Priority',
-    'FXSV' => 'FedEx Express saver (3 day)',
-    'USPP' => 'Postal Priority',
-    'GGRNDP' => 'Ground shipping',
-    'FXL1' => 'FedEx Freight Overnight (Air)',
-    'FXL2' => 'FedEx Freight 2-day (Air)',
-    'FXL3' => 'FedEx Freight 3-day (Air)',
-    'FXLE' => 'FedEx Freight Ground',
-   }.freeze
-
   DEFAULTS = {
-    ship_first_name: nil,
-    ship_last_name: nil,
-    ship_middle_initial: nil,
-    ship_address1: nil,
-    ship_address2: nil,
-    ship_address3: nil,
-    ship_city: nil,
-    ship_state: nil,
-    ship_post_code: nil,
-    ship_country_code: nil,
-    ship_phone1: nil,
-    ship_phone2: nil,
-    ship_fax: nil,
-    ship_email: nil,
-    ship_via: nil,
-    ship_request_date: nil,
-    ship_request_from: nil,
-    ship_request_warehouse: nil
-  }.freeze
+		ship_first_name:        nil,
+		ship_last_name:         nil,
+		ship_middle_initial:    nil,
+		ship_address1:          nil,
+		ship_address2:          nil,
+		ship_address3:          nil,
+		ship_city:              nil,
+		ship_state:             nil,
+		ship_post_code:         nil,
+		ship_country_code:      nil,
+		ship_phone1:            nil,
+		ship_phone2:            nil,
+		ship_fax:               nil,
+		ship_email:             nil,
+		ship_via:               nil,
+		ship_request_date:      nil,
+		ship_request_from:      nil,
+		ship_request_warehouse: nil
+  }
+
+  INTL_DEFAULTS = {
+    ship_no_later:     nil,
+    no_ship_before:    nil,
+    receive_no_later:  nil,
+    no_receive_before: nil,
+    dock_date:         nil,
+    ship_to_code:      nil
+  }.merge(DEFAULTS).freeze
 
   def defaults
-    DEFAULTS
+    IngramMicro.domestic_schema? ? DEFAULTS : INTL_DEFAULTS
   end
 
   def valid_shipping_methods?
     shipping_method = element[:ship_via]
-    !!SHIPPING_METHODS[shipping_method]
+    !!IngramMicro::SHIPPING_METHODS[shipping_method]
   end
 
   def shipping_method_name
     if valid_shipping_methods?
-      SHIPPING_METHODS[element[:ship_via]]
+      IngramMicro::SHIPPING_METHODS[element[:ship_via]]
```

Comment:
since `SHIPPING_METHODS` won't return an empty string we could combine this and `valid_shipping_methods?` in one method as:
```
  IngramMicro::SHIPPING_METHODS[element[:ship_via]] ||
    'Invalid shipping code'
```